src/test/java/bio/terra/service/resourcemanagement/MetadataDataAccessUtilsTest.java

```diff
@@ -0,0 +1,109 @@
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+@Category(Unit.class)
+public class MetadataDataAccessUtilsTest {
+
+  @InjectMocks private MetadataDataAccessUtils metadataDataAccessUtils;
+
+  @Mock private static ResourceService resourceService;
+
+  @Mock private static AzureBlobStorePdao azureBlobStorePdao;
+
+  @Mock private BlobContainerClientFactory blobContainerClientFactory;
+
+  @Mock private BlobSasUrlFactory blobSasUrlFactory;
+
+  private Dataset azureDataset;
+  private DatasetSummary azureDatasetSummary;
+  private BillingProfileModel defaultProfileModel;
+
+  @Before
+  public void setup() throws Exception {
+    UUID azureDatsetId = UUID.randomUUID();
+    UUID billingProfileModelId = UUID.randomUUID();
+    defaultProfileModel =
+        new BillingProfileModel().profileName("default profile").id(billingProfileModelId);
+    azureDatasetSummary =
+        new DatasetSummary()
+            .storage(
+                List.of(
+                    new AzureStorageResource(
+                        azureDatsetId,
+                        AzureCloudResource.STORAGE_ACCOUNT,
+                        AzureRegion.DEFAULT_AZURE_REGION)))
+            .defaultProfileId(billingProfileModelId)
+            .billingProfiles(List.of(defaultProfileModel));
+    DatasetTable sampleTable = new DatasetTable().id(UUID.randomUUID()).name("sample");
+    azureDataset =
+        new Dataset(azureDatasetSummary).tables(List.of(sampleTable)).name("test-dataset");
+
+    metadataDataAccessUtils = new MetadataDataAccessUtils(resourceService, azureBlobStorePdao);
+  }
+
+  @Test
+  public void testAzureAccessInfo() {
+    AzureStorageAccountResource storageAccountResource =
+        new AzureStorageAccountResource().resourceId(UUID.randomUUID()).name("michaelstorage");
+    when(resourceService.getStorageAccount(any(), any())).thenReturn(storageAccountResource);
+
+    when(azureBlobStorePdao.signFile(
+            any(),
+            any(),
+            eq("https://michaelstorage.blob.core.windows.net/metadata/parquet"),
+            eq(ContainerType.METADATA),
+            any(),
+            any()))
+        .thenReturn("https://michaelstorage.blob.core.windows.net/metadata/parquet/signedUrl");
+    when(azureBlobStorePdao.signFile(
+            any(),
+            any(),
+            eq("https://michaelstorage.blob.core.windows.net/metadata/parquet/sample"),
+            eq(ContainerType.METADATA),
+            any(),
+            any()))
+        .thenReturn(
+            "https://michaelstorage.blob.core.windows.net/metadata/parquet/sample/signedUrl");
+
+    AccessInfoParquetModel infoModel =
+        metadataDataAccessUtils.accessInfoFromDataset(azureDataset).getParquet();
+
+    Assert.assertEquals("test-dataset", infoModel.getDatasetName());
```

Comment:
You could replace these repeated values with variables, but it's a pretty small test so I'll leave that up to you!